### ISSUE TITLE
Forall: problem

### ISSUE THREAD

**Issue by rensink** [nl-utwente-groove/code#238]
There seems to be a problem with forall: nodes when using it in combination with creating new nodes.
I've attached an example demonstrating the problem: In this example, there's just one rule that tries to copy a node and all of its children.

The rule seems to match properly, but the result it creates is incorrect: The parent and its children are copied, but they aren't connected.
More precisely, for each child an empty node is created, and the child is connected to that empty node instead of the parent.

Don't know exactly when, but I know this bug has been around for at least a few weeks. \(I tried testing it on the latest revision just now, but Groove doesn't seem to recognize attr: nodes there anymore.. \) I also know that the bug wasn't there yet at revision 3606, if that helps.

Reported by: timmolderez

**Comment by rensink** [comment]
Forall example

Original comment by: timmolderez

**Comment by rensink** [comment]
This has been solved in the svn head \(since yesterday\).

Wrt the attr: problem: yes, we broke backward compatibility to accomodate better typing in the near future. You now have to type all data nodes explicitly \(use int:, bool: etc.\)

Backward compatibility was also broken for the subtyping support \(for the same reason\). For serious grammars, we want to promote the use of type graphs; this is from now on the only way to get subtyping.

Hope this doesn't cause too many problems...


Original comment by: rensink
